components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -128,13 +128,32 @@ pub fn year(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>>
     Ok(Some(Int::from(t.year())))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn day_name(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Bytes>> {
+    match t {
+        Some(t) => {
+            if t.invalid_zero() {
```

Comment:
The old implementation use `if t.is_zero() {` instead of `if t.invalid_zero() {`. Could you explain why do you want to change it?

Reply:
I tested in mysql and if I enter a non-zero invalid date, I get a warning.

mysql> select dayname('2019-00-00 00:00:00');
+--------------------------------+
| dayname('2019-00-00 00:00:00') |
+--------------------------------+
| NULL                           |
+--------------------------------+
1 row in set, 1 warning (0.00 sec)

Reply:
Nice finding! @iosmanthus PTAL